sdk/cosmos/azure_data_cosmos/src/pipeline/mod.rs

```diff
@@ -97,4 +100,93 @@ impl CosmosPipeline {
             }
         }))
     }
+
+    /// Helper function to read a throughput offer given a resource ID.
+    ///
+    /// ## Arguments
+    /// * `context` - The context for the request.
+    /// * `resource_id` - The resource ID to read the throughput offer for.
+    pub async fn read_throughput_offer(
+        &self,
+        context: Context<'_>,
+        resource_id: &str,
+    ) -> azure_core::Result<Option<Response<ThroughputProperties>>> {
+        #[derive(Model, Deserialize)]
+        struct OfferResults {
+            #[serde(rename = "Offers")]
+            pub offers: Vec<ThroughputProperties>,
+        }
+
+        // We only have to into_owned here in order to call send_query_request below,
+        // since it returns `Pager` which must own it's data.
+        // But in this case, we don't really _need_ the `Pager` to own it's data
+        // because we use it and dispose of it within the body of this method.
+        // If we wanted to optimize this later, we have a few options:
+        // 1. Give Pager a lifetime parameter so it can borrow it's context (proliferates lifetime parameters everywhere though...)
+        // 2. Don't use send_query_request. We expect the offer to be in the first page, so we could just make a regular request
+        // (but what if we get an empty page for some reason? is that something the server could do?)
```

Comment:
Given we discuss this in the issue, you might just link to it for full context instead of duplicating options here.

Reply:
Yep, we didn't have the issue when I wrote this. I'll update to link.